hazelcast/src/hazelcast/client/spi.cpp

```diff
@@ -1655,59 +1671,58 @@ namespace hazelcast {
                 }
 
                 bool
-                ClientPartitionServiceImpl::should_be_applied(const std::shared_ptr<connection::Connection> &connection,
-                                                              int32_t version,
+                ClientPartitionServiceImpl::should_be_applied(int32_t connection_id, int32_t version,
                                                               const std::vector<std::pair<boost::uuids::uuid, std::vector<int>>> &partitions,
                                                               const partition_table &current) {
                     auto &lg = client_.get_logger();
                     if (partitions.empty()) {
                         if (logger_.enabled(logger::level::finest)) {
-                            log_failure(connection, version, current, "response is empty");
+                            log_failure(connection_id, version, current, "response is empty");
                         }
                         return false;
                     }
-                    if (!current.connection || *connection != *current.connection) {
+                    if (!current.connection_id || connection_id != current.connection_id) {
                         HZ_LOG(lg, finest, 
-                            ([&current, &connection](){
-                                auto frmt = boost::format("Event coming from a new connection. Old connection: %1%, "
+                            ([&current, connection_id](){
+                                auto frmt = boost::format("Event coming from a new connection. Old connection id: %1%, "
                                                           "new connection %2%");
 
-                                if (current.connection) {
-                                    frmt = frmt % *current.connection;
+                                if (current.connection_id) {
+                                    frmt = frmt % current.connection_id;
```

Comment:
OK, we are only printing the connection id instead of the whole representation (`ClientConnection{alive=0 .. }`. But is this a good idea? It's just a monotonically increased (1, 2, 3,) number, right? Say I have seen this line in the log, I have no idea what this connection actually is. Unless I search in the log to find a full representation of the whole connection object, I don't know which member sent this event. And as far as I am aware, the whole connection representation is not printed on the log until the connection is closed. So there will be no way of understanding this log line before the connection is closed.

Reply:
Yes, it is monotonically increasing number. 

I understand the concern. Well, we can add the connection info onto the log where we print the auth result like at the end of the following line:
```
20/04/2021 19:12:45.223 INFO: [0x16f373000] hz.client_103[executor-test] [4.0.1] Authenticated with server  Address[localhost:5701]:55c2b3a8-95c9-497c-95d8-e9b863934bc4, server version: 4.1, local address: Address[127.0.0.1:62077]
```

But the user yes, still needs to look at the initial logs to match for the connection. Well, this was an idea to remove the connection pointer from the `cluster_view_listener::event_handler` and just use the integer connection id as the dependency but this is at the cost of not being as explicit log statement as before. It can still be resolved looking at the whole log. So the question is which is more preferable? The `weak_ptr` solution or no pointer but connection id solution.

Reply:
I think printing connection id is fine if we print the connection id immediately after a connection is made.

Reply:
Added the log.